pkg/datastore/filedb/filter.go

```diff
@@ -15,19 +15,177 @@
 package filedb
 
 import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
 	"github.com/pipe-cd/pipecd/pkg/datastore"
 )
 
-// TODO: Implement filterable interface for each collection.
 type filterable interface {
 	Match(e interface{}, filters []datastore.ListFilter) (bool, error)
 }
 
 func filter(col datastore.Collection, e interface{}, filters []datastore.ListFilter) (bool, error) {
 	fcol, ok := col.(filterable)
-	if !ok {
+	if ok {
+		return fcol.Match(e, filters)
+	}
+
+	// remarshal entity as map[string]interface{} struct.
+	raw, _ := json.Marshal(e)
+	var omap map[string]interface{}
+	if err := json.Unmarshal(raw, &omap); err != nil {
+		return false, err
+	}
+
+	for _, filter := range filters {
+		field := convertCamelToSnake(filter.Field)
+		if strings.Contains(field, ".") {
+			// TODO: Handle nested field name such as SyncState.Status.
+			return false, datastore.ErrUnsupported
+		}
+
+		val, ok := omap[field]
+		// If the object does not contain given field name in filter, return false immidiately.
+		if !ok {
+			return false, nil
+		}
+
+		cmp, err := compare(val, filter.Value, filter.Operator)
+		if err != nil {
+			return false, err
+		}
+
+		if !cmp {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func compare(val, operand interface{}, op datastore.Operator) (bool, error) {
+	var valNum, operandNum int64
+	switch v := val.(type) {
+	case int, int8, int16, int32, int64:
+		valNum = reflect.ValueOf(v).Int()
+	case uint, uint8, uint16, uint32, uint64:
+		valNum = reflect.ValueOf(v).Int()
```

Comment:
I think this should be `Uint()`.

Reply:
Nice catch 👍 In fact, I want to cast all numeric values listed in the switch case as `int64` to compare, but you're right, in case of the value of type `uint64` cast it to `int64` may lead to a wrong value, how do you think about removing the `uint64` (make it as unsupported value type) and keep the current casting: all to int64 type?

Reply:
I sounds good to me 👍